.test-infra/tools/stale_bq_datasets_cleaner.sh

```diff
@@ -33,11 +34,13 @@ for dataset in ${BQ_DATASETS[@]}; do
     if [[ $dataset =~ $template ]]; then
       # The BQ API reports LAST MODIFIED TIME in miliseconds, while unix works in seconds since epoch
       # thus why we need to convert to seconds.
-      LAST_MODIFIED_MS=`bq --format=json --project_id=$PROJECT show $dataset | jq -r .lastModifiedTime`
+      [[ `bq --format=json --project_id=$PROJECT show $dataset` =~ \"lastModifiedTime\":\"([0-9]+)\" ]]
+      LAST_MODIFIED_MS=${BASH_REMATCH[1]}
       LAST_MODIFIED=$(($LAST_MODIFIED_MS / 1000))
       if [[ $GRACE_PERIOD -gt $LAST_MODIFIED ]]; then
-        echo "Deleting $dataset (modified `date -d @$LAST_MODIFIED`) Command bq --project_id=$PROJECT rm -r -f $dataset"
-        bq --project_id=$PROJECT rm -r -f $dataset
+        echo "Deleting $dataset (modified `date -d @$LAST_MODIFIED`)"
+        # do not fail the script if delete dataset fail
+        bq --project_id=$PROJECT rm -r -f $dataset || true
```

Comment:
Hm, are we able to manually clean up the remaining datasets? Have you narrowed down which ones are the problem?

Reply:
This is the only part of the PR I'm not sure about, arguably we want the job to fail if this doesn't work - maybe we can catch the error, log the failing dataset, continue through the loop, then fail at the end?

Reply:
Yes, there are three of them: 
```
apache-beam-testing:beam_temp_dataset_2fce1dad3a414c7fb0d67a35ede80272
apache-beam-testing:beam_temp_dataset_36fabf70b9044847b7478c27fe6a6f46
apache-beam-testing:beam_temp_dataset_8828ff4eb5e9460e9ba05811b3f344ef
```
do not have permission to delete them manually either. Not sure how they got created. Once these 3 are removed, `|| true` can be removed also.

Reply:
<img width="415" alt="image" src="https://user-images.githubusercontent.com/8010435/210010421-0a65f825-71f6-4926-9950-1cf7f4afff76.png">

I suspect this happens when some user manually run an integration test whose account has some permission setting which prevents others from deleting it. Even if these dataset are removed, whenever some user adding running integration test, this task would fail again.

Reply:
Probably the right fix here is to give the service account a better permission set.

My worry if we just ignore this error is that then we entirely lose our signal. We should be failing the job and then manually removing the resources in that case. @lukecwik are you able to delete these? Looks like you may have the right permission set

- https://pantheon.corp.google.com/bigquery?q=search&referrer=search&page=dataset&p=apache-beam-testing&d=beam_temp_dataset_2fce1dad3a414c7fb0d67a35ede80272&project=apache-beam-testing&mods=dataflow_dev&orgonly=true&supportedpurview=organizationId
- https://pantheon.corp.google.com/bigquery?q=search&referrer=search&page=dataset&p=apache-beam-testing&d=beam_temp_dataset_36fabf70b9044847b7478c27fe6a6f46&project=apache-beam-testing&mods=dataflow_dev&orgonly=true&supportedpurview=organizationId&ws=!1m4!1m3!3m2!1sapache-beam-testing!2sbeam_temp_dataset_36fabf70b9044847b7478c27fe6a6f46
- https://pantheon.corp.google.com/bigquery?q=search&referrer=search&page=dataset&p=apache-beam-testing&d=beam_temp_dataset_8828ff4eb5e9460e9ba05811b3f344ef&project=apache-beam-testing&mods=dataflow_dev&orgonly=true&supportedpurview=organizationId&ws=!1m4!1m3!3m2!1sapache-beam-testing!2sbeam_temp_dataset_8828ff4eb5e9460e9ba05811b3f344ef

Reply:
Thanks @damccorm and  @lukecwik. The service account running this job is the same running other integration tests. I expect adding a dedicate service account would be technically more involved and has more safety concerns.

Reply:
Yep, that's fair - I more was wondering if we could update the existing service account to have these permissions (probably neither you nor me have the correct permission set to do so though).

In the short term, my vote would be to delete the bad datasets manually and then merge the PR without the `|| true`, or even better with some logic that does as much cleanup as it can and then fails the job if any of the cleanups failed. Longer term, if this is a persistent problem we could update the service account to handle this correctly.

Reply:
Yeah agree, for now let's fail the job at the end if a deletion if failed instead of just omit it. Will remove `|| true`